draft-ietf-tls-esni.md

```diff
@@ -1003,21 +1004,22 @@ client by computing its ServerHello as described here.
 The backend server begins by generating a message ServerHelloECHConf, which is
 identical in content to a ServerHello message with the exception that
 ServerHelloECHConf.random is equal to 24 random bytes followed by 8 zero bytes.
-It then computes a string
+It then computes an 8-byte string
 
 ~~~
-    accept_confirmation =
-        Derive-Secret(Handshake Secret,
-                      "ech accept confirmation",
-                      ClientHelloInner...ServerHelloECHConf)
+   accept_confirmation = HKDF-Expand-Label(0,
+      "ech accept confirmation",
+      Transcript-Hash(ClientHelloInner...ServerHelloECHConf),
+      8)
 ~~~
 
-where Derive-Secret and Handshake Secret are as specified in {{RFC8446}},
-Section 7.1, and ClientHelloInner...ServerHelloECHConf refers to the sequence of
-handshake messages beginning with the first ClientHello and ending with
+where HKDF-Expand-Label and Transcript-Hash are as defined in {{RFC8446}},
+Section 7.1, "0" indicates a string of Hash.length bytes set to zero, and
+ClientHelloInner...ServerHelloECHConf refers to the sequence of handshake
+messages beginning with the first ClientHello and ending with
```

Comment:
Not a problem with this PR, but...

I honestly don't understand this construction.  ServerHelloECHConf isn't part of the sequence listed in [Section 4.4.1 of RFC 8446](https://tools.ietf.org/html/rfc8446#section-4.4.1).  Maybe I can infer that it replaces ServerHello, but you need to *say* that.  And ClientHelloInner isn't listed there either.

I think that you need to be more careful about this part of the specification.

Reply:
Yeah, that's the idea. I updated the text to make this explicit.